### ISSUE TITLE
Updating a file where version has overridden `full_filename` fails

### ISSUE THREAD

**Issue by toby-brilliant** [carrierwaveuploader/carrierwave-aws#178]
**tl;dr:** When a file initially exists and that is updated, any versions with overridden `full_filename` will not make it out of the temp directory

My code looks like this:
```
class LogoUploader < CarrierWave::Uploader::Base
  def store_dir
    "uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
  end

  version :thumb
  version :small
  version :medium

  version :png do
    def full_filename(for_file = model.logo.file)
      "png_logo.png"
    end
  end
end
```

### Blank slate
```
> s.update!(logo: nil)
> s.logo.url
=> nil
```

```
aws s3 ls s3://dev.brilliantmade.com/uploads/store/logo/2076/
<nothing here>
```

### Uploading a file initially

```
> s.logo = File.open("app/assets/images/fulfillment/details_icon.png")
> s.logo.url
=> "https://s3.us-west-1.amazonaws.com/dev.brilliantmade.com/uploads/tmp/1699985556-433121775682744-0001-0914/details_icon.png"
```

```
aws s3 ls s3://dev.brilliantmade.com/uploads/tmp/1699985556-433121775682744-0001-0914/
2023-11-14 13:12:37     218814 details_icon.png
2023-11-14 13:12:40     218814 medium_details_icon.png
2023-11-14 13:12:40     218814 png_details_icon.png
2023-11-14 13:12:39     218814 small_details_icon.png
2023-11-14 13:12:38     218814 thumb_details_icon.png
```

Great success - the original image plus 4 versions are all in the temp directory.

```
> s.save!
> s.logo.url
=> "https://s3.us-west-1.amazonaws.com/dev.brilliantmade.com/uploads/store/logo/2076/details_icon.png"
```

```
aws s3 ls s3://dev.brilliantmade.com/uploads/store/logo/2076/
2023-11-14 13:14:36     218814 details_icon.png
2023-11-14 13:14:37     218814 medium_details_icon.png
2023-11-14 13:14:38     218814 png_logo.png
2023-11-14 13:14:37     218814 small_details_icon.png
2023-11-14 13:14:36     218814 thumb_details_icon.png
```
Everything is as it should be - the original image plus 4 versions are all in the permanent directory

### Updating the file
Starting from the step above, where the logo is populated:
```
> s.logo = File.open("/Users/tobymurray/Downloads/fulfillment_details.png")
> s.logo.url
=> "https://s3.us-west-1.amazonaws.com/dev.brilliantmade.com/uploads/tmp/1699985842-274372010805457-0002-5124/fulfillment_details.png"
```

```
aws s3 ls s3://dev.brilliantmade.com/uploads/tmp/1699985842-274372010805457-0002-5124/
2023-11-14 13:17:24     337881 fulfillment_details.png
2023-11-14 13:17:28     337881 medium_fulfillment_details.png
2023-11-14 13:17:28     337881 png_fulfillment_details.png
2023-11-14 13:17:27     337881 small_fulfillment_details.png
2023-11-14 13:17:26     337881 thumb_fulfillment_details.png
```
Everything is as it should be - the original image plus 4 versions are all in the temp directory

```
s.save!
```

```
aws s3 ls s3://dev.brilliantmade.com/uploads/store/logo/2076/

2023-11-14 13:18:44     337881 fulfillment_details.png
2023-11-14 13:18:46     337881 medium_fulfillment_details.png
2023-11-14 13:18:45     337881 small_fulfillment_details.png
2023-11-14 13:18:45     337881 thumb_fulfillment_details.png
```
Missing `png_fulfillment_details.png`.

**Comment by toby-brilliant** [comment]
Potentially related to the comment here: https://github.com/carrierwaveuploader/carrierwave-aws/issues/153#issuecomment-625942066
